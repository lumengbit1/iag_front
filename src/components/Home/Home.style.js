import styled from 'styled-components';

export const HomePage = styled.div`
  display: flex;
  justify-content: space-between;
  height: 100%;
`;

export const Title = styled.div`
  font-size: 30px;
  font-weight: bold;
  margin: 10px;
`;

export const PropertiesArea = styled.div`
  display: flex;
  flex-direction: column;
  justify-content: flex-start;
  align-items: center;
  padding-top: 5%;
  width: 49%;
  border: 1px solid black;
`;

export const Block = styled.div`
  border: 1px dashed #000;
  border-radius: 10px;
  padding: 20px;
  background-color: #fafbfa;
  min-width: 225px;
  min-height: 205px;
`;
