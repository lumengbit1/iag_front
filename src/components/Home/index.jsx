import React from 'react';
import PropertyList from '../PropertyList';
import {
  HomePage,
  PropertiesArea,
  Title,
  Block,
} from './Home.style';

const Home = () => (
  <HomePage>
    <PropertiesArea>
      <Title>
        Results
      </Title>

      <Block>
        <PropertyList type="results" />
      </Block>
    </PropertiesArea>

    <PropertiesArea>
      <Title>
        Saved Properties
      </Title>

      <Block>
        <PropertyList type="saved" />
      </Block>
    </PropertiesArea>
  </HomePage>
);

export default Home;
