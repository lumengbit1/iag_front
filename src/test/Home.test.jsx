import React from 'react';
import 'jest-styled-components';
import { Provider } from 'react-redux';
import thunk from 'redux-thunk';
import { render } from '@testing-library/react';
import { createStore, applyMiddleware } from 'redux';
import { combineReducers } from 'redux-immutable';
import Home from '../components/Home';
import eventList from '../reducers/reducer';

const store = createStore(combineReducers({ value: eventList }), applyMiddleware(thunk));

describe('Render Test', () => {
  it('case: expect rendering correct', () => {
    const { container } = render(
      <Provider store={store}>
        <Home />
      </Provider>,
    );

    expect(container).toMatchSnapshot();
  });
});
