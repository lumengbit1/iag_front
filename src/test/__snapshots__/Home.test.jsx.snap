// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Render Test case: expect rendering correct 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  height: 100%;
}

.c2 {
  font-size: 30px;
  font-weight: bold;
  margin: 10px;
}

.c1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-box-pack: start;
  -webkit-justify-content: flex-start;
  -ms-flex-pack: start;
  justify-content: flex-start;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  padding-top: 5%;
  width: 49%;
  border: 1px solid black;
}

.c3 {
  border: 1px dashed #000;
  border-radius: 10px;
  padding: 20px;
  background-color: #fafbfa;
  min-width: 225px;
  min-height: 205px;
}

<div>
  <div
    class="c0"
  >
    <div
      class="c1"
    >
      <div
        class="c2"
      >
        Results
      </div>
      <div
        class="c3"
      >
        <div
          data-testid="loading"
        >
          Loading...
        </div>
      </div>
    </div>
    <div
      class="c1"
    >
      <div
        class="c2"
      >
        Saved Properties
      </div>
      <div
        class="c3"
      >
        <div
          data-testid="loading"
        >
          Loading...
        </div>
      </div>
    </div>
  </div>
</div>
`;
